db/compaction/compaction_iterator.cc

```diff
@@ -214,6 +214,7 @@ bool CompactionIterator::InvokeFilterIfNeeded(bool* need_skip,
   CompactionFilter::Decision filter = CompactionFilter::Decision::kUndetermined;
   compaction_filter_value_.clear();
   compaction_filter_skip_until_.Clear();
+  blob_value_.Reset();
```

Comment:
Thanks so much for reporting and for the PR @sherriiiliu !

This issue is somewhat similar to what I'm addressing in #10490 w/r/t the DB iterator. I'm wondering if we could reset `blob_value_` in e.g. `CompactionIterator::Next()` instead of here? It would have the same benefit as in the DB iterator, namely we would release any cache handles as soon as possible. (Also note that GC uses `blob_value_` as well, so it might be non-empty for non-compaction filter related reasons.)

Reply:
Thanks @ltamasi for your suggestion. I have changed `blob_value_` reset to be in the beginning of `CompactionIterator::NextFromInput()` while loop when finding the next valid input